src/uwtools/drivers/upp.py

```diff
@@ -81,6 +100,40 @@ def driver_name(cls) -> str:
         """
         return STR.upp
 
+    @property
+    def output(self) -> OutputT:
+        """
+        Returns a description of the file(s) created when this component runs.
+        """
+        # Derive values from the current driver config. GRIB output filename suffixes include the
+        # forecast leadtime, zero-padded to at least 2 digits (more if necessary). Avoid taking the
+        # log of zero.
+        cf = self.config["control_file"]
+        leadtime = int(self.leadtime.total_seconds() / 3600)
+        suffix = ".GrbF%0{}d".format(max(2, int(log10(leadtime or 1)) + 1)) % leadtime
```

Comment:
Since it's only zero-padded to 2 digits (not 3), will this (or something like it) not work in Python?
```
suffix = f".GrbF{leadtime:02d}"
```
For example:
```
>>> leadtime = 1
>>> print(f".GrbF{leadtime:02d}")
.GrbF01
>>> leadtime = 10
>>> print(f".GrbF{leadtime:02d}")
.GrbF10
>>> leadtime = 100
>>> print(f".GrbF{leadtime:02d}")
.GrbF100
```

Reply:
Ooh, nice observation: We'd never have more than one leading zero. Let me give that a try.